src/__test__/pure_functions.test.js

```diff
@@ -0,0 +1,1012 @@
+          common: 'Ecuador',
+        },
+        ita: {
+          official: "Repubblica dell'Ecuador",
+          common: 'Ecuador',
+        },
+        jpn: {
+          official: 'エクアドル共和国',
+          common: 'エクアドル',
+        },
+        kor: {
+          official: '에콰도르 공화국',
+          common: '에콰도르',
+        },
+        nld: {
+          official: 'Republiek Ecuador',
+          common: 'Ecuador',
+        },
+        per: {
+          official: 'جمهوری اکوادور',
+          common: 'اکوادور',
+        },
+        pol: {
+          official: 'Ekwador',
+          common: 'Ekwador',
+        },
+        por: {
+          official: 'República do Equador',
+          common: 'Equador',
+        },
+        rus: {
+          official: 'Республика Эквадор',
+          common: 'Эквадор',
+        },
+        slk: {
+          official: 'Ekvádorská republika',
+          common: 'Ekvádor',
+        },
+        spa: {
+          official: 'República del Ecuador',
+          common: 'Ecuador',
+        },
+        swe: {
+          official: 'Republiken Ecuador',
+          common: 'Ecuador',
+        },
+        urd: {
+          official: 'جمہوریہ ایکوڈور',
+          common: 'ایکواڈور',
+        },
+        zho: {
+          official: '厄瓜多尔共和国',
+          common: '厄瓜多尔',
+        },
+      },
+      latlng: [
+        -2,
+        -77.5,
+      ],
+      landlocked: false,
+      borders: [
+        'COL',
+        'PER',
+      ],
+      area: 276841,
+      demonyms: {
+        eng: {
+          f: 'Ecuadorean',
+          m: 'Ecuadorean',
+        },
+        fra: {
+          f: 'Équatorienne',
+          m: 'Équatorien',
+        },
+      },
+      flag: '🇪🇨',
+      maps: {
+        googleMaps: 'https://goo.gl/maps/TbX8hUW4gcbRPZiK7',
+        openStreetMaps: 'https://www.openstreetmap.org/relation/108089',
+      },
+      population: 17643060,
+      gini: {
+        2019: 45.7,
+      },
+      fifa: 'ECU',
+      car: {
+        signs: [
+          'EC',
+        ],
+        side: 'right',
+      },
+      timezones: [
+        'UTC-06:00',
+        'UTC-05:00',
+      ],
+      continents: [
+        'South America',
+      ],
+      flags: [
+        'https://flagcdn.com/ec.svg',
+        'https://flagcdn.com/w320/ec.png',
+      ],
+    }));
+
+    expect(newState.countries).toEqual([]);
+    expect(newState.loading).toEqual('idle');
+    expect(newState.error).toEqual(null);
+  });
+});
+
+describe('fetchCountryData', () => {
+  jest.setTimeout(10000);
+  it('should dispatch the correct actions on successful API request', async () => {
+    // Mock the response data
+    const mockResponse = {
+      countries: [
+        {
+          name: {
+            common: 'Ecuador',
+            official: 'Republic of Ecuador',
+            nativeName: {
+              spa: {
+                official: 'República del Ecuador',
+                common: 'Ecuador',
+              },
+            },
+          },
+          tld: [
```

Comment:
- [ ] You have a failing test case here, It looks like you're encountering a timeout issue while running a test suite using Jest. The error message suggests increasing the timeout using jest.setTimeout(newTimeout). Kindly fix this
![image](https://github.com/malikhaiderkhan/Air-pollution/assets/81932478/287193e3-b01e-4eb5-9d1d-26f745b11c60)